File: backend/services/resort_service.dart
import '../models/resort.dart';

class ResortService {
  final List<Resort> _resorts = [];

  Future<List<Resort>> getAllResorts() async {
    return _resorts;
  }

  Future<Resort?> getResortById(String id) async {
    return _resorts.firstWhere((r) => r.id == id, orElse: () => null);
  }

  Future<Resort> createResort(String jsonString) async {
    // Very naive parsing; in real code use jsonDecode
    final resort = Resort(
      id: DateTime.now().millisecondsSinceEpoch.toString(),
      name: 'Sample Resort',
      location: 'Sample Location',
      pricePerNight: 100.0,
      imageUrl: 'https://example.com/image.jpg',
      description: 'A nice resort',
    );
    _resorts.add(resort);
    return resort;
  }

  Future<Resort?> updateResort(String id, String jsonString) async {
    final index = _resorts.indexWhere((r) => r.id == id);
    if (index == -1) return null;
    final resort = Resort(
      id: id,
      name: 'Updated Resort',
      location: 'Updated Location',
      pricePerNight: 150.0,
      imageUrl: 'https://example.com/updated.jpg',
      description: 'Updated description',
    );
    _resorts[index] = resort;
    return resort;
  }

  Future<bool> deleteResort(String id) async {
    return _resorts.removeWhere((r) => r.id == id) > 0;
  }
}

File: backend/services/resort_service.py
"""Resort service layer."""

from typing import List
from ..models.resort import ResortCreate, ResortOut
from ..models.resort import ResortBase

# In a real application this would interact with a database
# Here we use an in-memory list for demonstration

class ResortService:
    _resorts: List[ResortOut] = []
    _id_counter: int = 1

    async def get_all_resorts(self) -> List[ResortOut]:
        return self._resorts

    async def get_resort_by_id(self, resort_id: int) -> ResortOut | None:
        for resort in self._resorts:
            if resort.id == resort_id:
                return resort
        return None

    async def create_resort(self, resort_in: ResortCreate) -> ResortOut:
        resort = ResortOut(
            id=self._id_counter,
            name=resort_in.name,
            description=resort_in.description,
            location=resort_in.location,
            price_per_night=resort_in.price_per_night,
            amenities=resort_in.amenities,
        )
        self._resorts.append(resort)
        self._id_counter += 1
        return resort

# Additional methods for update/delete can be added
